Realm/RLMRealm.mm

```diff
@@ -277,6 +277,9 @@ + (SharedRealm)openSharedRealm:(Realm::Config const&)config error:(NSError **)ou
     try {
         return Realm::get_shared_realm(config);
     }
+    catch (SchemaMismatchException const& ex) {
+        throw;
```

Comment:
Rather than unconditionally throw here, I think you need to do the following:

```cpp
if (config.delete_realm_if_migration_needed) {
    throw;
} else {
    RLMRealmTranslateException(outError);
}
```